helix-core/src/movement.rs

```diff
@@ -29,15 +71,19 @@ pub fn move_horizontally(
             nth_next_grapheme_boundary(text, pos, count).min(end)
         }
     };
-    Range::new(if extend { range.anchor } else { pos }, pos)
```

Comment:
This is a common pattern, I'm planning to extract it into something like `range.transform(cond, pos)` I think.

```rust
range.transform(behaviour == SelectionBehaviour::Extend, pos)
```